nemo_curator/datasets/doc_dataset.py

```diff
@@ -37,6 +37,18 @@ def __len__(self) -> int:
     def persist(self) -> "DocumentDataset":
         return DocumentDataset(self.df.persist())
 
+    def repartition(
+        self,
+        divisions: Optional[List[int]] = None,
+        npartitions: Optional[int] = None,
+        partition_size: Optional[Union[int, str]] = None,
+        freq: Optional[str] = None,
+        force: bool = False,
+    ) -> "DocumentDataset":
+        return DocumentDataset(
+            self.df.repartition(divisions, npartitions, partition_size, freq, force)
+        )
```

Comment:
It might be worth exploring `functools.wraps` here to wrap df's repartition method. IIuc it might allow inheriting the function signature and docstrings, while not having to redefine all args.
According to LLM's
```python
    @wraps(dd.DataFrame.repartition)  # Ensures the docstring and signature are inherited
    def repartition(self, *args, **kwargs):
        return self.__class__(self.df.repartition(*args, **kwargs))
```

Reply:
Thanks!